q2_vsearch/tests/test_merge_pairs.py

```diff
@@ -87,11 +114,47 @@ def test_merge_pairs(self):
                 len(seq_lengths),
                 default_exp_sequence_counts[str(fastq_name)])
 
+    def test_unmerged_pairs(self):
+        with redirected_stdio(stderr=os.devnull):
+            _, unmerged = merge_pairs(self.input_seqs)
+
+        # manifest is as expected
+        self._test_manifest_unmerged(unmerged)
+
+        # expected number of fastq files are created
+        output_fastqs = list(unmerged.sequences.iter_views(FastqGzFormat))
+        self.assertEqual(len(output_fastqs), 6)
+
+        default_exp_sequence_counts = {
+            'BAQ2687.1_0_L001_R1_001.fastq.gz': 31,
+            'BAQ2687.1_0_L001_R2_001.fastq.gz': 31,
+            'BAQ3473.2_1_L001_R1_001.fastq.gz': 30,
+            'BAQ3473.2_1_L001_R2_001.fastq.gz': 30,
+            'BAQ4697.2_2_L001_R1_001.fastq.gz': 26,
+            'BAQ4697.2_2_L001_R2_001.fastq.gz': 26,
+        }
+        for fastq_name, fastq_path in output_fastqs:
+            seqs = skbio.io.read(
+                str(fastq_path),
+                format='fastq',
+                compression='gzip',
+                constructor=skbio.DNA
+            )
+            seqs = list(seqs)
+            seq_lengths = np.asarray([len(s) for s in seqs])
+            self._test_seq_lengths_unmerged(seq_lengths)
+
+            # expected number of sequences are unmerged
+            self.assertEqual(
+                len(seq_lengths),
+                default_exp_sequence_counts[str(fastq_name)]
+            )
+
     def test_merge_pairs_some_samples_w_no_joined_seqs(self):
         # minmergelen is set very high here, resulting in only one sequence
         # being merged across the three samples.
         with redirected_stdio(stderr=os.devnull):
-            obs = merge_pairs(self.input_seqs, minmergelen=279)
+            obs, _ = merge_pairs(self.input_seqs, minmergelen=279)
```

Comment:
This seems like a good place to check both the merged and unmerged files resulting from a single call to make sure that all input reads (by count) end up in one file or the other. I don't think you have that in one place elsewhere. I think that's a good test to include, whether it's in this specific test case or not.

Reply:
The `test_merge_pairs` and `test_unmerged_pairs` run the same command on the same data and assert the number of the merged and unmerged reads respectively. These numbers would sum to the number of records in the sample if one were to check. But this isn't necessary because the numbers come from vsearch directly.